src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -837,9 +793,10 @@ def forward(
         encoder_attention_mask: Optional[torch.Tensor] = None,
         layer_head_mask: Optional[torch.Tensor] = None,
         cross_attn_layer_head_mask: Optional[torch.Tensor] = None,
-        past_key_value: Optional[Tuple[torch.Tensor]] = None,
+        past_key_value: Optional[EncoderDecoderCache] = None,
```

Comment:
wrong type!

Reply:
Note that this is one-to-one the same as in `LLaMADecoderLayer`, where we can be sure any `past_key_value` is `Cache`:
https://github.com/huggingface/transformers/blob/3f93fd06949f9eae58e50fd0c9b8e60be82643bc/src/transformers/models/llama/modeling_llama.py#L688

Reply:
Good answer and on top of that we don't support tuples here